File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/SelectionActualMatchedCategorizer.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class SelectionActualMatchedCategorizer extends AbstractMatchedCategorizer {

    public SelectionActualMatchedCategorizer() {
        super("actualMatchedSelection_");
    }

    @Override
    public boolean isMarketSnapshotRequired() {
        return true;
    }

    @Override
    protected OptionalDouble getAmount(SettledBet bet) {
        Double matchedAmount = bet.getBetAction()
                .getMarketPrices()
                .getRunnerPrices(bet.getSelectionId())
                .getMatchedAmount();
        return matchedAmount == null ? OptionalDouble.empty() : OptionalDouble.of(matchedAmount);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/SelectionActualMatchedCategorizer.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.model.RealizedBet
import cz.fb.manaus.core.model.getRunnerPrices
import org.springframework.stereotype.Component

@Component
class SelectionActualMatchedCategorizer : AbstractMatchedCategorizer("actualMatchedSelection_") {

    override val isMarketSnapshotRequired: Boolean = true

    override fun getAmount(realizedBet: RealizedBet): Double? {
        return getRunnerPrices(realizedBet.betAction.runnerPrices, realizedBet.settledBet.selectionId).matchedAmount

    }

}
